toolkit/tools/internal/customizationmacros/customizationmacros.go

```diff
@@ -0,0 +1,133 @@
+		"# Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+	localeComments []string = []string{
+		"# This stops locale files from being installed. %%_install_langs acts as a filter for locales",
+		"# which start with the provides strings. Setting it to an invalid value (ie 'NONE') will",
+		"# prevent any locale files from being installed.",
+		"# To enable locale files, remove this file, or comment out '%%_install_langs <LOCALE STRING>'",
+		"# Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+)
+
+// AddCustomizationMacros adds the currently defined image custimization macros to the specified root directory.
+// For each of disableDocs and DisableRpmLocales a macro file is created with the corresponding macros defined in the
+// default rpm macros directory.
+func AddCustomizationMacros(rootDir string, disableDocs, disableRpmLocales bool, overrideRpmLocales string) (err error) {
+	if overrideRpmLocales != "" && disableRpmLocales {
+		return fmt.Errorf("cannot set both DisableRpmLocales and OverrideRpmLocales")
+	}
+
+	if disableDocs {
+		logger.Log.Debugf("Disabling documentation packages")
+		err = AddMacroFile(rootDir, rpm.DisableDocumentationDefines(nil), DisableDocsMacroFile, docComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable docs macro file:\n%w", err)
+		}
+	}
+	if disableRpmLocales {
+		logger.Log.Debugf("Disabling locale packages")
+		err = AddMacroFile(rootDir, rpm.DisableLocaleDefines(nil), ConfigureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable locales macro file:\n%w", err)
+		}
+	}
+	if overrideRpmLocales != "" {
+		logger.Log.Debugf("Overriding locale packages with (%s)", overrideRpmLocales)
+		err = AddMacroFile(rootDir, rpm.OverrideLocaleDefines(nil, overrideRpmLocales), ConfigureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add override locales macro file:\n%w", err)
+		}
+	}
+	return nil
+}
+
+// validateComments checks that the comments are valid for a macro file: ie they are empty or start with '#'
+func validateComments(comments []string) error {
+	for _, comment := range comments {
+		comment = strings.TrimSpace(comment)
+		if !strings.HasPrefix(comment, "#") && comment != "" {
+			return fmt.Errorf("extra comments must start with '#'(%s)", comment)
+		}
+	}
+	return nil
+}
+
+// AddMacroFile adds a macro file to the specified root directory with the specified macros. The macro file
+// is created in the default rpm macros directory. The macro file will include a default header, with additional comments
+// if desired. Each extra comment should start with a '#' character.
+func AddMacroFile(rootDir string, macros map[string]string, macroFileName string, extraComments []string) error {
+	const macrosPath = "/usr/lib/rpm/macros.d"
```

Comment:
To consider: it would be safer to retrieve this value from RPM itself in case it ever changes or we're building some unusual image. Unlikely, so not a high priority in my opinion.

How to get this from RPM:
```bash
rpm --eval %_rpmmacrodir
```

Reply:
Make this path more general, no need to limit to root.

Reply:
Ok, updated to preferentially use rpm --eval. It will fall back to the default value if rpm isn't there (useful for unit tests).